File: app/src/main/java/app/rpgbuilder/adapters/books/BooksAdapter.java
package app.rpgbuilder.adapters.books;

import static androidx.recyclerview.widget.RecyclerView.Adapter;
import static androidx.recyclerview.widget.RecyclerView.ViewHolder;

import static androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult;

import app.rpgbuilder.R;

import app.rpgbuilder.activities.BookActivity;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.content.Context;
import android.content.Intent;

import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import java.util.List;

public final class BooksAdapter extends Adapter<BookViewHolder> {

    @NonNull
    @Override
    public BookViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new BookViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.book_section, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull BookViewHolder holder, int position) {
        final Context context = holder.getBookIcon().getContext();

        final var appcompat = (AppCompatActivity) context;
        final var launcher = appcompat.registerForActivityResult(new StartActivityForResult(), data -> {
            final Intent intent = data.getData();
        });

        holder.getBookIcon().setImageResource(Book.BOOK_ICON);
        holder.getBookTitleAndContent().setText(BooksRepository.getBook(position).toString());

        holder.getBookView().setOnClickListener(view -> {
            final Intent intent = new Intent(context, BookActivity.class);

            intent.putExtra("title", holder.getBookTitle());
            intent.putExtra("content", holder.getBookContent());

            launcher.launch(new Intent(context, BookActivity.class));
        });
    }

    @Override
    public int getItemCount() {
        return BooksRepository.size();
    }
}

final class BookViewHolder extends ViewHolder {

    private final View bookView;

    private final ImageView bookIcon;

    private final TextView bookTitleAndContent;

    public BookViewHolder(View bookView) {
        super(bookView);

        this.bookView = bookView.findViewById(R.id.root_book_view);
        this.bookIcon = bookView.findViewById(R.id.book_icon);
        this.bookTitleAndContent = bookView.findViewById(R.id.book_title_content);
    }

    public View getBookView() {
        return bookView;
    }

    public ImageView getBookIcon() {
        return bookIcon;
    }

    public TextView getBookTitleAndContent() {
        return bookTitleAndContent;
    }

    public String getBookTitle() {
        return bookTitleAndContent.getText().toString().split("\n")[0];
    }

    public String getBookContent() {
        return bookTitleAndContent.getText().toString().split("\n")[1];
    }
}

File: app/src/main/java/app/rpgbuilder/tmpkt/activities/BooksAdapter.kt
package app.rpgbuilder.tmpkt.activities

import app.rpgbuilder.R

import app.rpgbuilder.activities.LibraryActivity

import android.content.Intent
import android.content.Context

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import android.widget.TextView
import android.widget.ImageView

import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult

import androidx.appcompat.app.AppCompatActivity

import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder

class BooksAdapter constructor(private val context: Context) : Adapter<BooksAdapter.BookViewHolder>() {

    inner class BookViewHolder constructor(view: View) : ViewHolder(view) {
        internal val bookView = view.findViewById<View>(R.id.root_book_view)
        internal val bookIcon = view.findViewById<ImageView>(R.id.book_icon)
        internal val bookTitleAndContent = view.findViewById<TextView>(R.id.book_title_content)

        internal val bookTitle = bookTitleAndContent.text.split("\n")[0]
        internal val bookContent = bookTitleAndContent.text.split("\n")[1]
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookViewHolder =
        BookViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.book_section, parent, false))

    override fun getItemCount(): Int =
        BooksRepository.size()

    override fun onBindViewHolder(holder: BookViewHolder, position: Int) {
        val appCompat = context as AppCompatActivity

        holder.bookIcon.setImageResource(BOOK_ICON)

        holder.bookTitleAndContent.text = "${BooksRepository.getBook(position).title}\n${BooksRepository.getBook(position).content}"

        holder.bookView.setOnClickListener {
            val intent = Intent(context, LibraryActivity::class.java)

            intent.putExtra("title", holder.bookTitle)
            intent.putExtra("content", holder.bookContent)

            appCompat.registerForActivityResult(StartActivityForResult()) {
            }.launch(Intent(context, MainActivity::class.java))
        }
    }
}